src/adapter/src/coord/ddl.rs

```diff
@@ -929,13 +929,51 @@ impl Coordinator {
         }
     }
 
+    /// Drops all pending replicas for a set of clusters
+    /// that are undergoing reconfiguration.
+    pub async fn drop_reconfiguration_replicas(&mut self, cluster_ids: BTreeSet<ClusterId>) {
+        let pending_replica_drops: Vec<Vec<DropObjectInfo>> = cluster_ids
+            .iter()
+            .map(|c| {
+                self.catalog()
+                    .get_cluster(c.clone())
+                    .replicas()
+                    .filter_map(|r| match r.config.location {
+                        ReplicaLocation::Managed(ref l) if l.pending => {
+                            Some(DropObjectInfo::ClusterReplica((
+                                c.clone(),
+                                r.replica_id,
+                                ReplicaCreateDropReason::Manual,
+                            )))
+                        }
+                        _ => None,
+                    })
+                    .collect::<Vec<DropObjectInfo>>()
+            })
+            .collect();
+        for replica_drops in pending_replica_drops {
+            self.catalog_transact(None, vec![Op::DropObjects(replica_drops)])
+                .await
+                .unwrap_or_terminate("cannot fail to drop replicas");
+        }
```

Comment:
Why do we do a single transaction per replica instead of a single transaction with all replicas?

Reply:
confusingly, if I flatten the above Vec of Vecs and do a single transaction there are a lot of test failures that seem completely unrelated. For example, the following test fails `./mzcompose --find testdrive run default blue-green.td` as well as `cargo test`.

Full test run:
https://buildkite.com/materialize/test/builds/88225#01914d7d-df74-4eba-ba28-33b389a503e0

I can't imagine a single connection will have more than cluster alter going at any given time, especially given that the execution is blocking. I'm pretty ok with leaving this a single transaction per cluster, but I am a bit confused and curious about why a single transaction for all replicas has this impact.